### ISSUE TITLE
Could not find function "run_de"

### ISSUE THREAD

**Issue by venchi9** [neurorestore/Libra#24]
Hello! 

I'm not really sure what I'm doing wrong.  I've successfully loaded Libra and can load the hagai_toy dataset.  When I try to run the DE I get this error:

head(hagai_toy@meta.data)
                      orig.ident nCount_RNA nFeature_RNA replicate label
2-CCAGCGACACAACTGT SeuratProject         26           11    mouse1  lps4
2-CAGCCGACAGATTGCT SeuratProject         20           12    mouse1  lps4
2-TGCACCTCAGTTCCCT SeuratProject         17            9    mouse1  lps4
2-CGTCCATTCCTATTCA SeuratProject         14           10    mouse1  lps4
2-GATGAAACACCAGGCT SeuratProject         15           10    mouse1  lps4
2-TCAATCTCATACAGCT SeuratProject          8            5    mouse1  lps4
                                                    cell_type
2-CCAGCGACACAACTGT bone marrow derived mononuclear phagocytes
2-CAGCCGACAGATTGCT bone marrow derived mononuclear phagocytes
2-TGCACCTCAGTTCCCT bone marrow derived mononuclear phagocytes
2-CGTCCATTCCTATTCA bone marrow derived mononuclear phagocytes
2-GATGAAACACCAGGCT bone marrow derived mononuclear phagocytes
2-TCAATCTCATACAGCT bone marrow derived mononuclear phagocytes

**> DE = run_de(hagai_toy)
Error in run_de(hagai_toy) : could not find function "run_de"**

I've tried re-installing Libra without any luck.

**Comment by jordansquair** [comment]
I do not think this is a Libra bug. Did you library(Libra) before running the code?

**Comment by ccasar** [comment]
I ran into this issue before when I accidentally installed the identically named [Libra package](https://cran.r-project.org/web/packages/Libra/index.html) from CRAN instead of installing from this github repo.

**Comment by horizon-mm** [comment]
Hi all, if you installed this package and executed update.packages() or update command from RStudio, etc, it will be replaced by the CRAN Libra that has a higher version number. You'll need to re-install this from github.

**Comment by Jorges1000** [comment]
Can you please change the name to avoid conflicts with CRA Libra? It was working just fine for me but after installing CRAN Libra as a dependency of another program stopped working. Had to remove CRAN Libra, but now I get this error installing neurorestore/Libra:

[1] "/shared/apps/R/4.2.0/lib/R/library"
Error: .onLoad failed in loadNamespace() for 'TMB', details:
  call: dyn.load(file, DLLpath = DLLpath, ...)
  error: unable to load shared object '/shared/apps/R/4.2.0/lib/R/library/TMB/libs/TMB.so':
  /usr/lib/x86_64-linux-gnu/libblas.so.3: undefined symbol: gotoblas
Execution halted
ERROR: lazy loading failed for package ‘Libra’
* removing ‘/shared/apps/R/4.2.0/lib/R/library/Libra’
Warning message:
In i.p(...) :
  installation of package ‘/tmp/RtmpS2W6BN/file15713dd59223/Libra_1.0.0.tar.gz’ had non-zero exit status

> sessionInfo()
R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.6 LTS

Matrix products: default
BLAS/LAPACK: /opt/OpenBLAS/lib/libopenblas_zenp-r0.3.21.so

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8        LC_COLLATE=C.UTF-8    
 [5] LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8    LC_PAPER=C.UTF-8       LC_NAME=C             
 [9] LC_ADDRESS=C           LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] Seurat_4.1.1.9003  data.table_1.14.2  sp_1.5-0           SeuratObject_4.1.1 workflowr_1.7.0   

loaded via a namespace (and not attached):
  [1] plyr_1.8.7            igraph_1.3.1          lazyeval_0.2.2        splines_4.2.0        
  [5] listenv_0.8.0         scattermore_0.8       usethis_2.1.5         ggplot2_3.3.6        
  [9] digest_0.6.29         htmltools_0.5.2       fansi_1.0.3           magrittr_2.0.3       
 [13] memoise_2.0.1         tensor_1.5            cluster_2.1.3         ROCR_1.0-11          
 [17] remotes_2.4.2         globals_0.15.0        matrixStats_0.62.0    spatstat.sparse_2.1-1
 [21] prettyunits_1.1.1     colorspace_2.0-3      ggrepel_0.9.1         xfun_0.31            
 [25] dplyr_1.0.9           callr_3.7.0           crayon_1.5.1          jsonlite_1.8.0       
 [29] progressr_0.10.1      spatstat.data_2.2-0   survival_3.3-1        zoo_1.8-10           
 [33] glue_1.6.2            polyclip_1.10-0       gtable_0.3.0          leiden_0.4.2         
 [37] pkgbuild_1.3.1        future.apply_1.9.0    abind_1.4-5           scales_1.2.0         
 [41] DBI_1.1.3             spatstat.random_2.2-0 miniUI_0.1.1.1        Rcpp_1.0.9           
 [45] viridisLite_0.4.0     xtable_1.8-4          reticulate_1.25       spatstat.core_2.4-2  
 [49] htmlwidgets_1.5.4     httr_1.4.3            RColorBrewer_1.1-3    ellipsis_0.3.2       
 [53] ica_1.0-2             pkgconfig_2.0.3       sass_0.4.1            uwot_0.1.11          
 [57] deldir_1.0-6          utf8_1.2.2            tidyselect_1.1.2      rlang_1.0.4          
 [61] reshape2_1.4.4        later_1.3.0           munsell_0.5.0         tools_4.2.0          
 [65] cachem_1.0.6          cli_3.3.0             generics_0.1.2        devtools_2.4.3       
 [69] ggridges_0.5.3        evaluate_0.15         stringr_1.4.0         fastmap_1.1.0        
 [73] yaml_2.3.5            goftest_1.2-3         processx_3.5.3        knitr_1.39           
 [77] fs_1.5.2              fitdistrplus_1.1-8    purrr_0.3.4           RANN_2.6.1           
 [81] pbapply_1.5-0         future_1.26.1         nlme_3.1-157          whisker_0.4          
 [85] mime_0.12             brio_1.1.3            compiler_4.2.0        rstudioapi_0.13      
 [89] plotly_4.10.0         curl_4.3.2            png_0.1-7             testthat_3.1.4       
 [93] spatstat.utils_2.3-1  tibble_3.1.7          bslib_0.3.1           stringi_1.7.6        
 [97] ps_1.7.0              desc_1.4.1            rgeos_0.5-9           lattice_0.20-45      
[101] Matrix_1.4-1          vctrs_0.4.1           pillar_1.7.0          lifecycle_1.0.1      
[105] spatstat.geom_2.4-0   lmtest_0.9-40         jquerylib_0.1.4       RcppAnnoy_0.0.19     
[109] cowplot_1.1.1         irlba_2.3.5           httpuv_1.6.5          patchwork_1.1.1      
[113] R6_2.5.1              promises_1.2.0.1      KernSmooth_2.23-20    gridExtra_2.3        
[117] parallelly_1.31.1     sessioninfo_1.2.2     codetools_0.2-18      MASS_7.3-57          
[121] assertthat_0.2.1      pkgload_1.2.4         rprojroot_2.0.3       withr_2.5.0          
[125] sctransform_0.3.4     mgcv_1.8-40           parallel_4.2.0        grid_4.2.0           
[129] rpart_4.1.16          tidyr_1.2.0           rmarkdown_2.14        Rtsne_0.16           
[133] git2r_0.30.1          getPass_0.2-2         shiny_1.7.1

**Comment by Jorges1000** [comment]
So, reinstalling glmmTMB resolved the problem, however to avoid further pain and allow simultaneous use of the CRAN package it would be great to change the name, perhaps "libra"?
